components/bluetooth/lib.rs

```diff
@@ -676,9 +685,13 @@ impl BluetoothManager {
             return drop(sender.send(Err(BluetoothError::Security)));
         }
         let services = self.get_gatt_services_by_uuid(&mut adapter, &device_id, &uuid);
+
+        // Step 2.7: If its argument is empty, throws a NotFoundError.
         if services.is_empty() {
             return drop(sender.send(Err(BluetoothError::NotFound)));
         }
```

Comment:
We should be able to remove this special case entirely, since we return the same value if we finish looping through services.